drivers/sensor/iis3dhhc/Kconfig

```diff
@@ -0,0 +1,75 @@
+# ST Microelectronics IIS3DHHC accelerometer sensor
+#
+# Copyright (c) 2019 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig IIS3DHHC
+	bool "IIS3DHHC accelerometer sensor"
+	depends on (SPI && HAS_DTS_SPI)
```

Comment:
`HAS_DTS_SPI` is selected by `SPI`, should not be needed.
And I think these should actually be cleaned up (follow up on https://github.com/zephyrproject-rtos/zephyr/pull/19050)

Reply:
@erwango 
I see. There is a bug #11064 which is fixed by 1634cf2.
Still not sure to understand why only SPI is selecting HAS_DTS_SPI.
Why I2C doesn't?

Reply:
Nittier subjective nitpick:

Can just make it `depends on SPI && HAS_DTS_API` btw.